src/Gui/Tree.cpp

```diff
@@ -1033,6 +1054,39 @@ void TreeWidget::onFinishEditing()
     }
 }
 
+// add childs of the selected tree object to selection
+void TreeWidget::onSelectAllChildren()
+{
+    // we only have this context menu entry if the selection is within one document
+    // so we can get the active document once
+    auto doc = Gui::Application::Instance->activeDocument()->getDocument();
+    auto docName = doc->getName();
+    // if the selected object is a document
+    if (this->contextItem && this->contextItem->type() == DocumentType) {

```

Comment:
Since contextItem can be an item of a non-active document the access to the active document will lead to incorrect behaviour.

You must access the correct document via contextItem:
* If it's a DocumentType you perform a cast to DocumentItem and get the Gui::Document directly.
* If it's an ObjectType you perform a cast to DocumentObjectItem and access the view provider and from there the Gui::Document.

Reply:
I thought by checking in advance that the function is only called if all selected items are within one document, I am safe.
I changed this now according to your suggestion.